### PR TITLE
Riverpod

### PR DESCRIPTION
**Pull request by jpnurmi**

### CODE REVIEW

File: packages/ubuntu_desktop_installer/lib/pages/network/ethernet_view.dart [hunk 1]
```diff
@@ -1,14 +1,13 @@
 import 'package:flutter/material.dart';
-import 'package:provider/provider.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
 import 'package:ubuntu_desktop_installer/l10n.dart';
+import 'package:ubuntu_desktop_installer/pages.dart';
```
Comment (jpnurmi) [inline packages/ubuntu_desktop_installer/lib/pages/network/ethernet_view.dart:4]: should be a local import of the network page only:
```dart
import 'network_page.dart';
```

File: packages/ubuntu_desktop_installer/lib/pages/network/hidden_wifi_view.dart [hunk 2]
```diff
@@ -1,17 +1,16 @@
 import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
 import 'package:form_field_validator/form_field_validator.dart';
-import 'package:provider/provider.dart';
 import 'package:ubuntu_desktop_installer/l10n.dart';
+import 'package:ubuntu_desktop_installer/pages.dart';
```
Comment (jpnurmi) [inline packages/ubuntu_desktop_installer/lib/pages/network/hidden_wifi_view.dart:5]: local import of the network page only

File: packages/ubuntu_desktop_installer/lib/pages/timezone/timezone_page.dart [hunk 3]
```diff
@@ -12,38 +12,26 @@ import 'package:yaru_widgets/yaru_widgets.dart';
...
-          create: (_) => TimezoneModel(getService<SubiquityClient>()),
-        ),
-        ChangeNotifierProvider(
-          create: (_) => TimezoneController(service: getService<GeoService>()),
-        ),
-      ],
-      child: const TimezonePage(),
-    );
-  }
+  static final timezoneModelProvider = ChangeNotifierProvider(
```
Comment (jpnurmi) [inline packages/ubuntu_desktop_installer/lib/pages/timezone/timezone_page.dart:18]: `TimezonePage.timezoneModelProvider` gets quite long and seems repetitive, is the "timezone" prefix intentional or did it slip in by accident? :)

**Review by jpnurmi — COMMENTED** [review]
Coming back to this with fresh eyes after a well-rested weekend, I already regret the choice of scoping providers within pages. Mostly because it forces some widgets to import the whole page when they just want to access the model...

File: packages/ubuntu_desktop_installer/lib/pages/timezone/timezone_page.dart (hunk 3 above)
Comment (d-loose) [reply packages/ubuntu_desktop_installer/lib/pages/timezone/timezone_page.dart]: That wasn't intentional - I renamed it :)

**Review by jpnurmi — COMMENTED** [review]
LGTM :+1: 

image

**Review by d-loose — APPROVED** [review]

**Pull request merged**